### PR TITLE
TF and GHA updates

### LINKED ISSUE #56: Add DivvyCloud action
Refs: https://github.com/byu-oit/hw-fargate-api/pull/92

### LINKED ISSUE #57: Use Terraform module version of Postman test lambda
It should simplify the codebase here.

Refs: https://github.com/byu-oit/hw-fargate-api/pull/68

### LINKED ISSUE #60: Comment Terraform Plan to PR
Refs: https://github.com/byu-oit/hw-lambda-api/blob/bf6d3ab65196759430bfbc21a26fcd9ad2da677c/.github/workflows/ci.yml#L231
Refs: https://github.com/byu-oit/hw-fargate-api/commit/41a66d6c93c5ff2a8ac56974ad424d77c9fea0f0

### PR DESCRIPTION
**Pull request by blakemorgan**
These are updates I noticed needed to happen the last time I used the module.

- Update Terraform version.
- Update Terraform modules and third-party GH actions
- Update Dependabot to the new GitHub version.
- Update TF provider constraints (it's what the previous `terraform0.13-updgrade` command did).
- Add Terraform lock files (introduced in 0.14, docs suggest to version them).
- Use Postman Lambda module instead of including it in the project.
- Add TF Plan analyzer
- Follow TF recommendation to use the `-json` flag with `terraform output` in CI pipelines.
- Follow NPM recommendation to use `npm ci` in CI pipelines.
- Cache NPM dependencies and use cache in pipelines

Closes #60.
Closes #57.
Closes #56.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -70,13 +74,20 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Set up Node.js
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2.1.4
```
Comment (yoshutch) [inline .github/workflows/ci.yml:77]: do we need the specific version or can we get away with `@v2`? (same with all setup-node steps below)

**Review by GaryGSC — COMMENTED** [review]
This is good stuff!

Nice job on the [caching](https://github.com/actions/cache/pull/69), too!

One nit: Now that we're using the Terraform version of the Postman Lambda, we no longer need to distinguish CI for the "Test Lambda" and CI for the "App Lambda". Can we call those `Lint App`, `Audit App`, and `Test App` for consistency with [hw-fargate-api](https://github.com/byu-oit/hw-fargate-api)?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (GaryGSC) [reply .github/workflows/ci.yml:77]: Dependabot, currently, tries to specify GitHub Actions down to the patch version as shown.

Comment (GaryGSC) [reply .github/workflows/ci.yml:77]: That's the argument I see in favor of specifying patch version. FWIW, I lean towards just `v2`, but I'm pretty ambivalent on this change.

Comment (blakemorgan) [reply .github/workflows/ci.yml:77]: Yep. I just did it because I knew that if I didn't, Dependabot would.

**Review by GaryGSC — APPROVED** [review]
Assuming Scott's concern is resolved, I'd say this LGTM.

**Pull request merged**